### PR TITLE
Add pull request guidelines to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by gtritchie**
## Summary

- Adds a "Pull request guidelines" section to CONTRIBUTING.md covering local build/test expectations, a two-open-PR limit for external contributors, and guidance on AI-assisted contributions
- Tidies grammar and Posit style guide compliance in the surrounding sections (sentence fragment fix, em-dashes removed, "and/or" dropped, bold-for-emphasis removed, etc.)

## Test plan

- [ ] Render CONTRIBUTING.md on GitHub and confirm formatting

### CODE REVIEW

**Review by kevinushey — APPROVED** [review]
LGTM -- thanks for putting this together.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,7 +30,37 @@ We welcome contributions to RStudio. To submit a contribution:
...
+Every pull request is expected to have been built and tested locally before it is opened. In the PR description, please include:
+
+- **What you tested**: the scenarios, workflows, or behaviors you verified, including any edge cases you considered
+- **Which platforms you tested on**: macOS, Windows, or Linux
+
+If you were unable to test on a particular platform, say so explicitly. We'd much rather know up front than discover it during review or after merge.
+
+### Limit on Open Pull Requests
+
+External contributors may have at most two open pull requests at a time. Additional PRs opened while two are already open will be closed without review.
```
Comment (MariaSemple) [inline CONTRIBUTING.md]:
```suggestion
We ask that external contributors keep at most two open pull requests at a time. Additional PRs opened while two are already open will be closed without review, but may be re-opened at later time.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,7 +30,37 @@ We welcome contributions to RStudio. To submit a contribution:
...
+
+Once a contributor has established a track record of thoughtful, high-quality contributions, we're happy to lift this limit on a case-by-case basis. Just ask.
```
Comment (MariaSemple) [inline CONTRIBUTING.md]:
```suggestion
Once a contributor has established a track record of thoughtful, high-quality contributions, we're happy to lift this limit on a case-by-case basis.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,7 +30,37 @@ We welcome contributions to RStudio. To submit a contribution:
...
+
+### AI-Assisted Contributions
+
+Using AI tools to help write code or PR descriptions is not prohibited, but the responsibility for the contribution is yours. Before submitting, please:
+
+- Read every line of the change and make sure you understand it
+- Confirm the change actually addresses a real problem, not a hypothetical one inferred from the code
+- Verify the change fits the surrounding code style and conventions
+- Make sure the PR description matches what the diff actually does
```
Comment (MariaSemple) [inline CONTRIBUTING.md:61]: I'm not 100% set on this one, what do you think?
```suggestion
- Make sure the PR description matches what the diff actually does
- Include an acknowledgement that AI was used to help write this code 
```

Comment (gtritchie) [reply CONTRIBUTING.md:61]: I think it's a given that AI is being used on most if not all PRs. I'd almost be more worried about a change made without AI at this point.
